packages/next-swc/crates/next-core/src/next_server/resolve.rs

```diff
@@ -80,7 +80,15 @@ impl ResolvePlugin for ExternalCjsModulesResolvePlugin {
         if *condition(self.root).matches(context).await? {
             return Ok(ResolveResultOption::none());
         }
-        if !matches!(&*request.await?, Request::Module { .. }) {
+        let request_value = &*request.await?;
+        if !matches!(request_value, Request::Module { .. }) {
+            return Ok(ResolveResultOption::none());
+        }
+
+        // from https://github.com/vercel/next.js/blob/8d1c619ad650f5d147207f267441caf12acd91d1/packages/next/src/build/handle-externals.ts#L188
```

Comment:
Can you add another comment on the webpack side so that if someone changes them they don't accidentally forget to change this one.